fix(dispatch-service): fix temporary directories for containers

jonrkarr opened the pull request:
Closes #4281
Closes #4262 

With this change, I was able to run simulations with Smoldyn in UConn HPC.

I now see this in the log, but it doesn't seem to prevent execution
```
_XSERVTransmkdir: Owner of /tmp/.X11-unix should be set to root
```

jonrkarr committed:
fix(api): handled older simulation specifications that lack styles

jonrkarr committed:
refactor(api): applied similar change to other attributes of simulation specifications

bilalshaikh42 committed:
refactor(api): clean health check jobs after run

biosimulations-daemon committed:
chore(release): 9.7.2 [skip ci]

## [9.7.2](https://github.com/biosimulations/biosimulations/compare/v9.7.1...v9.7.2) (2022-02-19)

### Bug Fixes

* **api:** handled older simulation specifications that lack styles ([01d7829](https://github.com/biosimulations/biosimulations/commit/01d78290d1f6c5710c47b7005531e5b483371101))

renovate-bot committed:
chore(deps): update dependency cypress to v9.5.0

jonrkarr committed:
fix(dispatch-service): fix temporary directories for containers

jonrkarr commented:
If this doesn't work, we might need Mike's help.

renovate-bot committed:
chore(deps): update eslint

jonrkarr committed:
perf(combine-api): added ability to directly parse OMEX manifest files

jonrkarr committed:
fix(dispatch-service): corrected capitalization of SED-ML, COMBINE

jonrkarr committed:
perf(dispatch-service): switched to directly parsing manifests

biosimulations-daemon committed:
chore: format files. [skip ci]

Signed-off-by: BioSimulations-Daemon <biosimulationsdaemon@gmail.com>

biosimulations-daemon committed:
chore(release): 9.7.3 [skip ci]

## [9.7.3](https://github.com/biosimulations/biosimulations/compare/v9.7.2...v9.7.3) (2022-02-19)

### Bug Fixes

* **dispatch-service:** corrected capitalization of SED-ML, COMBINE ([e267376](https://github.com/biosimulations/biosimulations/commit/e267376e9c56d5ad59537b2f1d4378ebdf4138cf))

### Performance Improvements

* **combine-api:** added ability to directly parse OMEX manifest files ([a43a50e](https://github.com/biosimulations/biosimulations/commit/a43a50ebb5738bb97bf8483148af1c9ca04bcbf0))
* **dispatch-service:** switched to directly parsing manifests ([b86f48c](https://github.com/biosimulations/biosimulations/commit/b86f48c017106d79bd63ac0f53e8b5207db18bfd))

jonrkarr committed:
perf(combine-api): sped up JSON encoding

biosimulations-daemon committed:
chore: format files. [skip ci]

Signed-off-by: BioSimulations-Daemon <biosimulationsdaemon@gmail.com>

biosimulations-daemon committed:
chore(release): 9.7.4 [skip ci]

## [9.7.4](https://github.com/biosimulations/biosimulations/compare/v9.7.3...v9.7.4) (2022-02-19)

### Performance Improvements

* **combine-api:** sped up JSON encoding ([44fe6c1](https://github.com/biosimulations/biosimulations/commit/44fe6c13f8c2fd4e621ec12bb5b3f4a99ee87ddf))

bilalshaikh42 commented on apps/dispatch-service/src/app/services/sbatch/sbatch.service.ts:
```diff
@@ -257,23 +262,23 @@ srun --job-name="Save-outputs-to-S3" \
-    --include '*.h5' \
-    --content-type 'application/hdf5'\
-    --acl public-read
+    --endpoint-url ${storageEndpoint} \
```
Are these whitespace changes intentional/tested? I remember that this section was very sensitive to spacing, but not exactly what the issue was, but only the first step of the upload would succeed

jonrkarr replied:
Just to make it more readable. Seemed to work fine.

bilalshaikh42 commented on apps/dispatch-service/src/app/services/sbatch/sbatch.service.ts:
```diff
@@ -205,17 +206,21 @@ echo -e '${cyan}========================================== Downloading COMBINE/O
+executeArchiveStatus=$?
+rm -rf \${TEMP_DIRNAME}
+if [[ $executeArchiveStatus -ne 0 ]]; then
+  exit $executeArchiveStatus
```
Will this prevent any logs from being uploaded if present in the event that the execution fails?

jonrkarr committed:
fix(combine-api): fixed JSON serialization for Swagger app

jonrkarr committed:
fix(combine-api): fixed JSON serialization for Swagger app by replacing ujson with orjson

jonrkarr committed:
test(combine-api): added tests for Swagger UI and OpenAPI spec resolution

jonrkarr committed:
refactor(combine-api): renamed maxPolicy to minPolicy, returned to standard json encoding

jonrkarr committed:
docs(combine-api): clarified documentation of classes and attributes for alg substitution

jonrkarr committed:
refactor(combine-api): removed unused import

jonrkarr committed:
refactor(combine-api): increased gunicorn log level

jonrkarr committed:
perf(combine-api): decreasing number of requests in between restarting

jonrkarr committed:
refactor(combine-api): reapply orjson

jonrkarr committed:
chore(combine-api): prettified Pipfile.locK

jonrkarr committed:
feat(combine-api): upgraded Brian2, pyNeuroML, NEURON

biosimulations-daemon committed:
chore: update sitemaps [skip ci]

Signed-off-by: BioSimulations-Daemon <biosimulationsdaemon@gmail.com>

biosimulations-daemon committed:
chore(release): 9.8.0 [skip ci]

# [9.8.0](https://github.com/biosimulations/biosimulations/compare/v9.7.4...v9.8.0) (2022-02-22)

### Bug Fixes

* **combine-api:** fixed JSON serialization for Swagger app ([6d59f90](https://github.com/biosimulations/biosimulations/commit/6d59f901844d8bbfd69d50fe3fda09de19a77d92))
* **combine-api:** fixed JSON serialization for Swagger app by replacing ujson with orjson ([e10c291](https://github.com/biosimulations/biosimulations/commit/e10c2911d94ad5dee1cfb0ea6f27ac4d7f059fa6))

### Features

* **combine-api:** upgraded Brian2, pyNeuroML, NEURON ([329e07a](https://github.com/biosimulations/biosimulations/commit/329e07aae60292b02321ececee7d6768d26e87df))

### Performance Improvements

* **combine-api:** decreasing number of requests in between restarting ([438c6fb](https://github.com/biosimulations/biosimulations/commit/438c6fb14bf25b51a74ea1f3dd28e76a945548a1))

biosimulations-daemon committed:
chore(release): cleaned up changelog [skip ci]

jonrkarr committed:
fix(dispatch-service): corrected error tracking of S3 uploads, always trying to upload results to S3

jonrkarr committed:
fix(api): fixed path to output archives

jonrkarr commented:
Replaced by #4303
